### ISSUE TITLE
Nested extender alternates parents spec does not respect :root

### ISSUE THREAD

**Issue by davidkpiano** [sass/sass#1641]
**Issue:** The [nested extender alternates parents](https://github.com/sass/sass/blob/a128adaaf789310b12c64b8e7675b36fb1bc6462/test/sass/extend_test.rb#L599) specification does not respect the `:root` pseudo-class, for which there exists the inferred constraint that there can be no deeper ancestor as it is _always_ the root of the document, [according to the W3C specification](http://www.w3.org/TR/selectors/#root-pseudo). Given this, ancestors of `:root`, albeit semantically valid, will _never_ be selected, as they can't exist, and thus should be excluded from the alternated parent permutations.

``` scss
:root .foo {
  color: red;
}

.bar .baz {
  @extend .foo;
}
```

**EXPECTED:**

``` scss
:root .foo, :root .bar .baz {
  color: red;
}
```

**ACTUAL:**

``` scss
// '.bar :root .baz' will never exist
:root .foo, :root .bar .baz, .bar :root .baz {
  color: red;
}
```

**Comment by davidkpiano** [comment]
Spec assertion, if you guys want:

``` ruby
def test_nested_extender_respects_root
  assert_extends(':root .foo', '.bar .baz {@extend .foo}',
    ':root .foo, :root .bar .baz')
end
```

**Comment by nex3** [comment]
Fixed by cc3a41a40cfb5e974e7b2e57da5fec0e2be382fb.
